tests/by-util/test_rm.rs

```diff
@@ -676,6 +676,36 @@ fn test_remove_inaccessible_dir() {
     scene.ucmd().arg("-rf").arg(dir_1).succeeds();
     assert!(!at.dir_exists(dir_1));
 }
+#[test]
+#[cfg(not(windows))]
```

Comment:
what is the behavior under windows?

Reply:
I don't have access to windows unfortunately, i reckon the behavior should be same with just the '/' being '\' in windows .

Regarding the behavior on windows, i don't know if an stat call is made to check whether the directory exists before printing out to stderr. I can presume it is the same as unix with the stat call first .

Basically GNU does the stat call if the file doesn't exist, it exits with code 0. If the file exists and fits our pattern, we print to stderr and exit with code 1. I wonder if we should do the pattern check first and exit with code 1 if it matches or else do a stat call later essentially deviating from gnu in lieu for better performance.

Reply:
Feel free to pull stuff from my PR. I assume yours will get merged in cause yours was first anyways, and it was your issue. My tests included Windows, and on the CI passed, so I assume my code and tests all worked on Windows. I figured it was just changing from a '/' -> '\'

Reply:
@just-an-engineer Sorry for the wasted effort, i forgot to link the issue while creating the PR. I will put you as a coauthor, if this gets merged.

Reply:
No worries. That's why I linked it above. I wouldn't say no the brownie points, but it's not a big deal. I think once you move it away from regex, it won't be too different anyways. But like I said, I think the _only_ real difference is the OS string delimiter is '\', while on Unix it's '/'.
Take a look at my test file first. Notice how I had the Windows vs Linux `del` var for formatting. I think I'd actually go back and change my `rm` code to use that as well, instead of having 2 large conditional blocks, you can have a single block, and make the conditional compilations just 1 line each. From there, at compiletime, I'd think the compiler is smart enough (or hopefully will be, if not we can raise an issue in rust-lang if we want) to make it into a single string constant. If you're reworking from regex to strings, I'd def recommend going that route. Conditionally set the delimiter, then one single block of code for both Unix and Windows

Reply:
Or, if you prefer, I can make you coauthor on mine, and we can submit that since it seems to do what sylvestre is asking already

Reply:
I just made you the coauthor on those commits in the other PR. I figure either that will get merged, or yours will and it won't matter anyways. I also changed up my code to use that simplified schema that I describe, if you want to pull inspiration from it. Or not